test/common/common/log_macros_test.cc

```diff
@@ -118,6 +120,83 @@ TEST(Logger, checkLoggerLevel) {
   EXPECT_THAT(test_obj.executeAtTraceLevel(), testing::Eq(2));
 }
 
+void spamCall(std::function<void()>&& call_to_spam, const uint32_t num_threads) {
+  std::vector<std::thread> threads(num_threads);
+  std::promise<void> signal_all_threads_running;
+  std::shared_future<void> future(signal_all_threads_running.get_future());
+
+  std::atomic<uint32_t> runcount = 0;
+  for (auto& thread : threads) {
+    thread = std::thread([future, &call_to_spam, &runcount] {
+      runcount++;
+      future.wait();
+      call_to_spam();
+    });
+  }
+  // Allow threads to accrue on future.wait() to maximize concurrency on the call
+  // we are testing.
+  while (runcount != num_threads) {
```

Comment:
What about something like https://abseil.io/docs/cpp/guides/synchronization#barrier?

Reply:
Ooh, that's awesome. Switched to use `absl::Barrier` in de23f2a